File: Medium/Triplets with Sum in Range/triplets-with-sum-in-range.py
class Solution:

    def countTriplets(self, arr: list[int], l: int, r: int) -> int:
        # code here
        arr.sort()
        n = len(arr)
        res = 0
        res2 = 0
        for i in range(n - 2):
            sum = 0

            s = i + 1
            e = n - 1

            while s<e:
                sum = arr[s] + arr[e] + arr[i]
                if sum <= r:
                    res += e - s
                    s += 1
                else:
                    e -= 1

        for i in range(n - 2):
            sum = 0
            s = i + 1
            e = n - 1
            while s<e:
                sum = arr[s] + arr[e] + arr[i]
                if sum<l:
                    res2 += e - s
                    s += 1
                else:
                    e -= 1

        return res - res2

File: Medium/Triplets with Sum in Range/triplets-with-sum-in-range.cpp
class Solution {
  public:
  int countTriplets(vector<int> &arr, int l, int r) {
        sort(begin(arr), end(arr));
        int n = arr.size();
        int res = 0 , res2 = 0;
        for (int i = 0 ; i < n-1 ; i++){
            int s = i+1;
            int e = n-1;
            
            while (s < e){
                int sum = arr[i]+arr[s]+arr[e];
                
                if (sum <= r){
                    res += e-s;
                    s++;
                }else e--;
            }
        }
        for (int i = 0 ; i < n-1 ; i++){
            int s = i+1;
            int e = n-1;
            
            while (s < e){
                int sum = arr[i]+arr[s]+arr[e];
                if (sum < l){
                    res2 += e-s;
                    s++;
                }else e--;
            }
        }
        return res-res2;
    }
};